components/builder-api/src/server/resources/channels.rs

```diff
@@ -443,19 +444,10 @@ async fn promote_package(req: HttpRequest,
                 &auditevent,
                 &*conn,
             ) {
-                Ok(_) => {
-                    if let Some(bus) = &state.eventbus.as_ref() {
-                        let event_type = EventType::PackageChannelMotion;
-                        match BuilderEvent::new(event_type, None, json!(&auditevent)) {
-                             Ok(event) => bus.inner.publish(event).await,
-                             Err(err) => debug!("Failed to create EventBus event for {}: {}", event_type, err)
-                        }
-                    }
-                }
+                Ok(_) => publish_event!(&state, EventType::PackageChannelMotion, json!(&auditevent)),
```

Comment:
:thinking: Is `publish_event` actually a function of `AppState`?

Reply:
While you're at it with the macro, you could stuff the `json!` call _inside_ `publish_event!`, which would simplify this callsite.

Reply:
Agreed on both accounts. Handled in https://github.com/habitat-sh/builder/pull/1540/commits/172a3f954170aa79ba01ecca30f6117774a3e19f